packages/docker-engine/docker-engine.spec

```diff
@@ -54,8 +54,7 @@ export VERSION=%{gover}
 export GITCOMMIT=%{gitrev}
 export BUILDTIME=$(date -u -d "@%{source_date_epoch}" --rfc-3339 ns 2> /dev/null | sed -e 's/ /T/')
 export PLATFORM="Docker Engine - Community"
-chmod +x ./hack/make/.go-autogen
-./hack/make/.go-autogen
+source ./hack/make/.go-autogen
```

Comment:
Why is this sourced instead of being run?

Reply:
I followed what the [`binary`](https://github.com/moby/moby/blob/2773f81aa5e9e34733675a7aa7e4219391caccb0/hack/make/.binary#L15) script does in the moby repo.

Reply:
`./hack/make/.go-autogen` [changed significantly](https://github.com/moby/moby/commit/675b414f56063826457cbc8f0c165af67198bfa8#diff-28a29ae1b36a23b6fd072d1ab3bd970e743b200728fe8a2598ea14277193025b) between 19.03.15 and 20.10.4 and the result is that building Docker this way will lose a bunch of the version information we expect.

In addition to the `source` change that you made, we'll also need to update the `go build` line to include `${LDFLAGS}`.

Reply:
I updated the `go build` command to include the `LDFLAGS`. I can see the commit and version in the docker logs:

```
Jun 17 19:41:00  dockerd[1651]: time="2021-06-17T19:41:00.748861055Z" level=info msg="Docker daemon" commit=363e9a88a11be517d9e8c65c998ff56f774eb4dc graphdriver(s)=overlay2 version=20.10.4
```

The log line uses the dockerversion paramters passed in `LDFLAGS`:

https://github.com/moby/moby/blob/b9ad7b96bd86e5f632e5fb4f36f98dcc145014fc/daemon/daemon.go#L1141

Reply:
Can you post the output of `docker version` too?

Reply:
The output of `docker version`:

```
bash-5.0# docker version
Client: Docker Engine - Community
 Version:           20.10.4
 API version:       1.41
 Go version:        go1.16.3
 Git commit:        d3cb89ee53287cf9214ad5fbc9a111190f607c72
 Built:             Tue Apr 18 14:29:00 2017
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.4
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.16.3
  Git commit:       363e9a88a11be517d9e8c65c998ff56f774eb4dc
  Built:            Sat Mar 16 00:40:00 2013
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.4.4+bottlerocket
  GitCommit:        05f951a3781f4f2c1911b05e61c160e9c30eaa8e
 runc:
  Version:          1.0.0-rc93+bottlerocket
  GitCommit:        12644e614e25b05da6fd08a38ffa0cfe1903fdec
 docker-init:
  Version:          0.19.0
  GitCommit:

```